native/cpp/CommonCpp/CryptoTools/CryptoModule.cpp

```diff
@@ -149,18 +98,20 @@ bool CryptoModule::prekeyDoesntExist() {
 }
 
 bool CryptoModule::isPrekeySignatureValid() {
-  const std::string signingPublicKey =
-      getSigningPublicKey(this->getIdentityKeys());
-  const std::string prekey = parseOLMPrekey(this->getPrekey());
-  const std::string preKeySignature = this->getPrekeySignature();
-
-  const OlmBuffer prekeyBytes = Base64::decode(prekey);
   try {
-    this->verifySignature(signingPublicKey, prekeyBytes, preKeySignature);
+    folly::dynamic identityKeysObj = folly::parseJson(this->getIdentityKeys());
+    std::string signingPublicKey = identityKeysObj["ed25519"].asString();
```

Comment:
Why'd we get rid of `getSigningPublicKey`? We seem to be doing the same thing here, but I guess with less robust error handling. I guess we didn't need the error handling here?